src/Compilers/Core/Portable/MetadataReader/MetadataHelpers.cs

```diff
@@ -1049,17 +1056,150 @@ public static bool TryDecode(ImmutableArray<byte> bytes)
         }
 
         /// <summary>
+        /// Returns true if the Unicode character can be a part of a C# identifier.
+        /// </summary>
+        /// <param name="ch">The Unicode character.</param>
+        private static bool IsIdentifierPartCharacter(char ch)
```

Comment:
Please replace all of this with an inline test that the character is an ascii letter or digit.